pkg/resource/deploy/step_generator.go

```diff
@@ -1921,6 +1957,46 @@ func (sg *stepGenerator) GenerateDeletes(targetsOpt UrnTargets, excludesOpt UrnT
 	return steps, nil
 }
 
+// This function is called by the deployment executor in response to a AdditionalStepsEvent.
+func (sg *stepGenerator) HandleAdditionalSteps(event AdditionalStepsEvent) ([]Step, error) {
+	steps := event.Steps()
+	// TODO actually validate the steps
+	// return sg.validateSteps(steps)
+	return steps, nil
+}
+
+// getViews returns the set of views for a given URN.
+func (sg *stepGenerator) getViews(urn resource.URN) []plugin.View {
+	var views []plugin.View
+	for _, res := range sg.deployment.prev.Resources {
+		if res.ViewOf == urn {
+			view := plugin.View{
+				Type:    res.URN.Type(),
+				Name:    res.URN.Name(),
+				Inputs:  res.Inputs,
+				Outputs: res.Outputs,
+			}
+			if res.Parent != "" && res.Parent != urn {
+				view.ParentType = res.Parent.Type()
+				view.ParentName = res.Parent.Name()
+			}
+			views = append(views, view)
+		}
+	}
+	return views
```

Comment:
Nit: Is `deployment.prev.Resources` mutated during a deployment? If not, maybe we can build a `map[URN][]plugin.View`s once and use that instead of this (apologies if this is already on the TODO list and just not done for expediency of sharing here).